tests/frameworks/gtest/include/gtest/gtest.h

```diff
@@ -295,7 +296,7 @@ class GTEST_API_ AssertionResult {
   // assertion's expectation). When nothing has been streamed into the
   // object, returns an empty string.
   const char* message() const {
-    return message_.get() != NULL ?  message_->c_str() : "";
+    return success_ ? "" : (message_.get() != NULL ?  message_->c_str() : "");
```

Comment:
nit: this is kind of gross

I guess you could change it to (!success_ && message_.get()) ? message_->c_str() : "";   ? does that even look better?